docs/src/modules/utils/useCustomizationPlayground.tsx

```diff
@@ -0,0 +1,355 @@
+  [k: string]: { examples: CustomizationItemsType; slots: string[] };
+};
+
+export interface UseCustomizationPlaygroundProps {
+  examples: PickersSubcomponentType;
+  children?: React.ReactElement | React.ReactElement[];
+  componentName: string;
+}
+export const customizationLabels: CustomizationLabelType = {
+  customTheme: 'Custom Theme',
+  styledComponents: 'Styled Components',
+  sxProp: 'SX Prop',
+};
+
+export const DEFAULT_COLORS = { deepPurple, amber, pink, blue, grey };
+export type ColorKey = keyof typeof DEFAULT_COLORS;
+
+export type StyleTokensType = {
+  color: ColorKey;
+  padding: number;
+  borderRadius: number;
+  borderWidth: number;
+};
+const styleTokens: StyleTokensType = {
+  color: 'deepPurple',
+  padding: 1,
+  borderRadius: 2,
+  borderWidth: 1,
+};
+export type HandleTokenChangeType = (
+  token: keyof Partial<typeof styleTokens>,
+  value: string | keyof typeof DEFAULT_COLORS | number,
+) => void;
+
+export type UseCustomizationPlaygroundReturnType = {
+  selectedDemo: string | null;
+  customizationOptions: Partial<CustomizationLabelType> | null;
+  selectedCustomizationOption: keyof CustomizationLabelType | null;
+  selectDemo: (interactionTarget: string | null) => void;
+  setSelectedCustomizationOption: (
+    customizationOption: keyof CustomizationLabelType | null,
+  ) => void;
+  selectedSlot: string | null;
+  setSelectedSlot: (slot: string | null) => void;
+  codeExample: string | null;
+  availableSlots: string[] | null;
+  handleTokenChange: HandleTokenChangeType;
+  selectedTokens: StyleTokensType;
+  selectedExample?: CustomizationItemType | null;
+};
+
+export function withStyles(
+  Component: React.ElementType,
+  selectedTokens: StyleTokensType,
+  selectedCustomizationOption: string | null,
+  selectedDemo: string | null,
+  selectedSlot: string | null,
+) {
+  return function StyledChild<T extends BoxProps>(props: T) {
+    const theme = useTheme();
+
+    const tokens = {
+      padding: selectedTokens.padding,
+      borderRadius: selectedTokens.borderRadius,
+      borderColor: DEFAULT_COLORS[selectedTokens.color][500],
+      border: `${selectedTokens.borderWidth}px solid`,
+      backgroundColor:
+        theme.palette.mode === 'light'
+          ? DEFAULT_COLORS[selectedTokens.color][100]
+          : DEFAULT_COLORS[selectedTokens.color][700],
+      color:
+        theme.palette.mode === 'light'
+          ? DEFAULT_COLORS[selectedTokens.color][900]
+          : DEFAULT_COLORS[selectedTokens.color][100],
+    };
+
+    if (selectedCustomizationOption === 'sxProp') {
+      const sxProp = {
+        [`& .Mui${selectedDemo}-${selectedSlot}`]: { ...tokens },
+      };
+      return <Component {...props} sx={{ ...sxProp, ...props?.sx }} />;
+    }
+
+    if (selectedCustomizationOption === 'customTheme') {
+      const newTheme = createTheme({
+        components: {
+          [`Mui${selectedDemo}`]: { styleOverrides: { [selectedSlot as string]: { ...tokens } } },
+        },
+      });
```

Comment:
It should take the parent theme into account:

<img width="418" alt="Screenshot 2023-09-17 at 19 37 36" src="https://github.com/mui/mui-x/assets/3165635/d05acae5-6574-4646-bd23-1dfad2ff79cf">

See https://mui.com/material-ui/customization/theming/#system-ThemeNestingExtend.tsx